### ISSUE TITLE
Programmatically reference the temp directory used by the test framework

### ISSUE THREAD

**Issue by lblackstone** [pulumi/pulumi#3868]
I couldn't find a way to get a reference to the temporary directory used by the test framework during the `ExtraRuntimeValidation` step.

The use case here was that the program created some files relative to the project directory, but there was no simple way to access that path during validation.

**Comment by lunaris** [comment]
I don't think this has bitten us/been referenced with any severity for a long time, so going to close this off as stale.
